sdk/file_sdk/XMPFileFormat/xmp_helper.h

```diff
@@ -0,0 +1,52 @@
+#ifndef XMP_HELPER_H
+#define XMP_HELPER_H
+
+#define TXMP_STRING_TYPE std::string
+#define XMP_INCLUDE_XMPFILES 1
+
+#include "xmpio_over_istream.h"
+#include <XMP.incl_cpp>
+#include <XMP.hpp>
+#include <time.h>
+#include <string_utils.h>
+#include <mutex>
+#include "tag.h"
+
+using std::string;
+using std::vector;
+using std::shared_ptr;
+
+namespace {
+const XMP_StringPtr kMsipNamespace = "http://www.microsoft.com/msip/1.0/";
+const XMP_StringPtr kMsipLabelNamespace = "http://www.microsoft.com/msip/label/1.0/";
+const XMP_StringPtr kMsipLabelExtendedNamespace = "http://www.microsoft.com/msip/label/extended/1.0/";
+const XMP_StringPtr kLabelsArrayName = "labels";
+const XMP_StringPtr kExtendedStructName = "extended";
+}
+
+namespace mip {
+namespace file {
+
+class XMPHelper {
+
+public:
+  static string GetStringByKey(const SXMPMeta& metadata, const string &labelsItemPath, const XMP_StringPtr fieldName, const XMP_StringPtr alternativeFieldName, const XMP_StringPtr oldestFieldName = nullptr);
+
+  static vector<Tag> Deserialize(const SXMPMeta& metadata);
+
+  static vector<Tag> DeserializeObsoleteFormat(const SXMPMeta& metadata);

```

Comment:
>static vector<Tag> DeserializeObsoleteFormat(const SXMPMeta& metadata); [](start = 2, length = 71)

we can not support the obsolete format in the sdk.

Reply:
even after you saw that there are a lot of labeled files with the old format?

---
In reply to: [138017608](https://github.com/AzureAD/rms-sdk-for-cpp/pull/140#discussion_r138017608) [](ancestors = 138017608)

Reply:
It was not "a lot"...
Yes.

---
In reply to: [138031592](https://github.com/AzureAD/rms-sdk-for-cpp/pull/140#discussion_r138031592) [](ancestors = 138031592,138017608)